src/app/mesh/qgsmaptooleditmesh.cpp

```diff
@@ -0,0 +1,806 @@
+/***************************************************************************
+  qgsmaptooleditmesh.cpp - QgsMapToolEditMesh
+
+ ---------------------
+ begin                : 24.6.2021
+ copyright            : (C) 2021 by Vincent Cloarec
+ email                : vcloarec at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgsmaptooleditmesh.h"
+
+#include "qgisapp.h"
+
+#include "qgsdoublespinbox.h"
+#include "qgsmeshdataprovider.h"
+#include "qgsmapcanvas.h"
+#include "qgsmapmouseevent.h"
+#include "qgsmessagebar.h"
+#include "qgsmeshlayer.h"
+#include "qgsmesheditor.h"
+#include "qgstriangularmesh.h"
+#include "qgsrubberband.h"
+#include "qgssnapindicator.h"
+#include "qgsvertexmarker.h"
+#include "qgsguiutils.h"
+
+
+QgsZValueWidget::QgsZValueWidget( const QString &label, QWidget *parent ): QWidget( parent )
```

Comment:
not worth to have ui file? (not sure)

Reply:
no, for now, only one spinbox and one label